jib-core/src/main/java/com/google/cloud/tools/jib/cache/Cache.java

```diff
@@ -115,6 +116,30 @@ public CachedLayer writeUncompressedLayer(
         uncompressedLayerBlob, LayerEntriesSelector.generateSelector(layerEntries));
   }
 
+  /**
+   * Retrieves the cached manifest for an image reference.
+   *
+   * @param imageReference the image reference
+   * @return the manifest for the image reference, if found
+   * @throws IOException if an I/O exception occurs
+   */
+  public Optional<ManifestTemplate> retrieveManifest(ImageReference imageReference)
+      throws IOException {
+    return cacheStorageReader.retrieveManifest(imageReference);
+  }
+
+  /**
+   * Retrieves the cached container configuration for an image reference.
+   *
+   * @param imageReference the image reference
+   * @return the container configuration for the image reference, if found
+   * @throws IOException if an I/O exception occurs
+   */
+  public Optional<ContainerConfigurationTemplate> retrieveContainerConfiguration(
```

Comment:
For Manifest V2.2 (and OCI too I believe), it should return both the manifest and the container config at the same time in a single transaction due to concurency.

Reply:
What do you mean by this? Return them both from the same method/using the same lock?

Reply:
Yeah, think about this, where Process A wants to get V2.2 manifest and its container config.

1. Process A calls `retrieveManifst()`.
1. Process B calls `writeMetadata(differentManifest, differentContainerConfig)`.
1. Process A calls `retrieveContainerConfig()`.

Then the container config at step 3 is for the new manifest from step 2, which may be out of sync with the manifest 1 in step 1.